src/app-layer-htp-file.c

```diff
@@ -186,13 +186,17 @@ int HTPFileOpenWithRange(HtpState *s, HtpTxUserData *txud, const uint8_t *filena
     uint8_t *keyurl;
     uint32_t keylen;
     if (tx->request_hostname != NULL) {
-        keylen = bstr_len(tx->request_hostname) + filename_len;
+        uint32_t hlen = (uint32_t)bstr_len(tx->request_hostname);
+        if (bstr_len(tx->request_hostname) > UINT16_MAX) {
```

Comment:
should we even accept this at all? A hostname shouldn't be this long and even if it is a 64k limit seems too high?

Reply:
What would be the good limit ?

Reply:
isn't there a 256 limit on hostnames? Don't recall with certainty

Reply:
https://curl.se/docs/CVE-2023-38545.html :-p